ddtrace/tracer/span.go

```diff
@@ -251,12 +251,10 @@ func (s *span) SetUser(id string, opts ...UserMonitoringOption) {
 		delete(root.Meta, keyUserID)
 		idenc := base64.StdEncoding.EncodeToString([]byte(id))
 		trace.setPropagatingTag(keyPropagatedUserID, idenc)
-		s.context.updated = true
```

Comment:
I think removing this can have side effects in `propagatorW3c.injectTextMap`:

```go
	if ctx.updated ||
		(ctx.trace != nil && !strings.HasPrefix(ctx.trace.propagatingTag(tracestateHeader), "dd=")) ||
		ctx.trace.propagatingTagsLen() == 0 {
		writer.Set(tracestateHeader, composeTracestate(ctx, p, ctx.trace.propagatingTag(tracestateHeader)))
	} else {
		writer.Set(tracestateHeader, ctx.trace.propagatingTag(tracestateHeader))
	}
	return nil
```

Is it a requirement for your feature to remove these usages?

Reply:
So I essentially had the context as always being flagged to being `updated` as `p:` will change whenever there was a span created from an extracted context and needs to recompose the `tracestate`

I think the side-effect of removing `updated` is that recomposing the tracestate will always occur which is the expected effect of that.

If that makes sense

Otherwise I can re-add the `updated` field, but it'll be a rare occurrence as I see it for `updated` to be `false`